crates/red_knot_python_semantic/src/types/mro.rs

```diff
@@ -76,21 +76,14 @@ impl<'db> Mro<'db> {
             // This *could* theoretically be handled by the final branch below,
             // but it's a common case (i.e., worth optimizing for),
             // and the `c3_merge` function requires lots of allocations.
-            [single_base] => {
-                let single_base = ClassBase::try_from_ty(*single_base, db).ok_or(*single_base);
-                single_base.map_or_else(
-                    |invalid_base_ty| {
-                        let bases_info = Box::from([(0, invalid_base_ty)]);
-                        Err(MroErrorKind::InvalidBases(bases_info))
-                    },
-                    |single_base| {
-                        let mro = std::iter::once(ClassBase::Class(class))
-                            .chain(single_base.mro(db))
-                            .collect();
-                        Ok(mro)
-                    },
-                )
-            }
+            [single_base] => ClassBase::try_from_ty(db, *single_base).map_or_else(
+                || Err(MroErrorKind::InvalidBases(Box::from([(0, *single_base)]))),
+                |single_base| {
+                    Ok(std::iter::once(ClassBase::Class(class))
+                        .chain(single_base.mro(db))
+                        .collect())
+                },
+            ),
```

Comment:
Nit: I always find `map_or_else` somewhat hard to read because the else becomes before the map (huh?)

```suggestion
            [single_base] => ClassBase::try_from_ty(db, *single_base)
                .map(|single_base| {
                    std::iter::once(ClassBase::Class(class))
                        .chain(single_base.mro(db))
                        .collect()
                })
                .ok_or_else(|| MroErrorKind::InvalidBases(Box::from([(0, *single_base)]))),
```

Reply:
I don't really have a strong preference either way. I weakly prefer `map_or_else`, but I agree with you that the argument order is kind of confusing. I won't object if you push another PR switching to this idiom ;)